### ISSUE TITLE
tests FileReadToEndNotReadable keeps failing on release/v1.25

### ISSUE THREAD

**Issue by noah8713** [envoyproxy/envoy#25614]
**If you are reporting *any* crash or *any* potential security issue, *do not*
open an issue in this repo. Please report the issue via emailing
envoy-security@googlegroups.com where the issue will be triaged appropriately.**

*Title*: tests FileReadToEndNotReadable keeps failing

*Description*:
> bazel.release //test/... --test_env=ENVOY_IP_TEST_VERSIONS=v4only --flaky_test_attempts=9
FAILED: //test/common/filesystem:filesystem_impl_test (Summary)
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test.log
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test_attempts/attempt_1.log
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test_attempts/attempt_2.log
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test_attempts/attempt_3.log
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test_attempts/attempt_4.log
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test_attempts/attempt_5.log
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test_attempts/attempt_6.log
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test_attempts/attempt_7.log
      /build/tmp/_bazel_root/3712b2d6d3ae88430b07e4b942dcf6a8/execroot/envoy/bazel-out/k8-opt/testlogs/test/common/filesystem/filesystem_impl_test/test_attempts/attempt_8.log

[  FAILED  ] 1 test, listed below:
[  FAILED  ] FileSystemImplTest.FileReadToEndNotReadable

test/common/filesystem/filesystem_impl_test.cc:118: Failure
Expected: file_system_.fileReadToEnd(file_path) throws an exception of type EnvoyException.
  Actual: it throws nothing.

*Call Stack*:
> test/common/filesystem/filesystem_impl_test.cc:118: Failure
Expected: file_system_.fileReadToEnd(file_path) throws an exception of type EnvoyException.
  Actual: it throws nothing.
Stack trace:
  0x4dfac5: Envoy::Filesystem::FileSystemImplTest_FileReadToEndNotReadable_Test::TestBody()
  0x9bdc8c: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x9bdb4d: testing::Test::Run()
  0x9bead7: testing::TestInfo::Run()
... Google Test internal frames ...

```
INFO: 31886 processes: 12062 internal, 1 local, 19822 processwrapper-sandbox, 1 worker.
//test/common/filesystem:filesystem_impl_test                            FAILED in 9 out of 9 in 0.1s

```
As a part of https://github.com/envoyproxy/envoy/pull/24398, this change was introduced by reverting original path not exist
```
EXPECT_THROW(file_system_.fileReadToEnd(TestEnvironment::temporaryPath("envoy_this_not_exist")),
               EnvoyException);
```
I checked [unit tests runs ](https://github.com/envoyproxy/envoy/actions/runs/3714238000/jobs/6297882381)were skipped 

```
Run ./mobile/tools/should_run_ci.sh
  ./mobile/tools/should_run_ci.sh
  shell: /bin/bash -e {0}
Skipping unittests because there are no mobile changes on [2](https://github.com/envoyproxy/envoy/actions/runs/3714238000/jobs/6297882381#step:3:2)4[3](https://github.com/envoyproxy/envoy/actions/runs/3714238000/jobs/6297882381#step:3:3)98/merge
```
 
envoy branch : release/v1.25
build image: envoy-build-ubuntu:7304f974de2724617b7492ccb4c9c58cd420353a

If needed can we amend envoy_this_not_exist. Not sure of original intent and hence didnt raise PR. Please advice

**Comment by phlax** [comment]
@noah8713 i have a feeling this may have been fixed on `main` and not yet backported

cc @ravenblackx

**Comment by noah8713** [comment]
Thanks. Double checked main doesn't have it , ref https://github.com/envoyproxy/envoy/blob/main/test/common/filesystem/filesystem_impl_test.cc#L119 . Proposed https://github.com/envoyproxy/envoy/pull/25641 PTAL

**Comment by ravenblackx** [comment]
The breaking change was making the test functional at all. The proposed fix (reverting that part of the change) just makes the test duplicate the behavior of the test above it, while falsely pretending to test something else. (And reduces coverage for the environment in which the test works as expected.)

I suppose what's happening here is the file system in which the release test takes place does not support setting the file to unreadable (or still allows reading it afterwards), so the read doesn't fail. Removing this test entirely as unreliable would be better than recreating the prior lie (either way reduces coverage the same).

**Comment by phlax** [comment]
im trying to understand some things here (other than the actual test/bug)

is this a problem on `main` ? 

why does this test not fail in CI for 1.25 branch ?

**Comment by phlax** [comment]
...looking at PR and the PR is against `main` not the 1.25 branch so im guessing that means the problem is present there also

**Comment by noah8713** [comment]
> ...looking at PR and the PR is against `main` not the 1.25 branch so im guessing that means the problem is present there also

@phlax @ravenblackx  thanks for review. yes its correct as I checked it wasnt in main branch too and didnt knew the original intent of the same so fixed it accordingly. Not sure we can add a note/TODO in current PR if you guys are ok so that @ravenblackx concern can be addressed that in future whoever improvises FS UTs can refer. Thoughts ?

**Comment by phlax** [comment]
> I checked [unit tests runs ](https://github.com/envoyproxy/envoy/actions/runs/3714238000/jobs/6297882381)were skipped

re this - that is mobile CI - i think this test _is_ being run in the azure pipelines - im still trying to understand why i havent seen this fail anywhere (iirc)

**Comment by phlax** [comment]
cc @RyanTheOptimist  If we can land this today it can be included in the pending release (#25828) - not sure if that matters as its seemingly test only but given the title i would rather include if possible

/backport

**Comment by phlax** [comment]
arggh - commented on the issue not the PR ...
